wp-content/plugins/ig-wp-api-extensions/endpoints/v2/RestApi_ModifiedContent.php

```diff
@@ -0,0 +1,330 @@
+		 * We should implement at least a half-fledged query builder to cope with the different needs
+		 * (or use an adequate framework).
+		 */
+		$select = $this->build_query_select();
+		$from = $this->build_query_from($initial_event);
+		$where = $this->build_query_where();
+		$groups = $this->build_query_groups();
+		$order_clauses = $this->build_query_order_clauses();
+
+		$where_first = array_shift($where);
+		$where_rest = $where;
+		// filter the recurring events when the JOIN translations does not
+		if( isset($initial_event) ) {
+			$where_rest[] = "em_events.recurrence_id = {$initial_event->event_id}";
+		}
+
+		return
+			"SELECT $select
+			$from
+			WHERE $where_first " .
+			($where_rest ? "AND " . join(" AND ", $where_rest) : "") . " " .
+			($groups ? "GROUP BY " . join(",", $groups) : "") . " " .
+			($order_clauses ? "ORDER BY " . join(",", $order_clauses) : "");
+	}
+
+	/**
+	 * @return string
+	 */
+	protected function build_query_select() {
+		return "posts.ID, posts.post_title, posts.post_type, posts.post_status, posts.post_modified_gmt,
+					posts.post_excerpt, posts.post_content, posts.post_parent, posts.menu_order, posts.guid,
+					users.user_login, usermeta_firstname.meta_value as author_firstname, usermeta_lastname.meta_value as author_lastname";
+	}
+
+	/**
+	 * @param array $initial_event
+	 * @return string
+	 */
+	protected function build_query_from($initial_event = null) {
+		global $wpdb;
+		$current_language = ICL_LANGUAGE_CODE;
+		// when the recurring events of an initial event are selected, join on the element_id of the initial event because recurring events don't have their own entries in the translations table
+		return "FROM $wpdb->posts posts" . ( isset($initial_event) ? "
+				JOIN {$wpdb->prefix}icl_translations translations
+						ON translations.element_type = 'post_event-recurring'
+						AND translations.element_id = '{$initial_event->ID}'
+						AND translations.language_code = '$current_language'" : "
+				JOIN {$wpdb->prefix}icl_translations translations
+						ON translations.element_type = 'post_{$this->current_request->post_type}'
+						AND translations.element_id = posts.ID
+						AND translations.language_code = '$current_language'" ) . "
+				JOIN $wpdb->users users
+						ON users.ID = posts.post_author
+				JOIN $wpdb->usermeta usermeta_firstname
+						ON usermeta_firstname.user_id = users.ID
+						AND usermeta_firstname.meta_key = 'first_name'
+				JOIN $wpdb->usermeta usermeta_lastname
+						ON usermeta_lastname.user_id = users.ID
+						AND usermeta_lastname.meta_key = 'last_name'";
+	}
+
+	/**
+	 * @return array
+	 */
+	protected function build_query_where() {
+		$since = $this->current_request->rest_request->get_param('since');
+		if(strtotime($since) < strtotime(self::FORCE_UPDATE_DATE)) {
+			//if the last update is after the deadline, pull all content by setting the since date to the beginning of 2015
+			$since = "2015-01-01T00:00:00+02:00";
```

Comment:
I think we should set $since to self::FORCE_UPDATE_DATE in this case, shouldn't we?

Reply:
No, actually we want 2015-something